### DISCUSSION TITLE
How to define/transfer the cognition.run code into html script so that it works on other platforms?

### DISCUSSION THREAD

**Discussion by lana-mrkonja** [jspsych/jsPsych#2557] [Q&A] [answered]
Unfortunately, my institution does not support cognition.run so I need to transfer the experiment written directly on Cognition.run to another experiment running platform. But I'm having trouble with defining the experiment using HTML instead of Javascript only - and I am not even sure if that's all that needs to be done (is there some other part of the code on Cognition.run that communicates with cognition website maybe that also needs to be removed?). 

Note: I am (brand) new to jsPsych so I don't really know what the code structure works. The code was written by another PhD student that's not part of the project anymore, and it seems that he also wasn't familiar with jspsych prior to coding the experiment.

**Comment by rankiejiang** [comment, marked as answer]
Hi
you can check out the demo of jspsych 👍 https://www.jspsych.org/7.2/tutorials/hello-world/

Usually, I save my cognition code into js file,
and then I add this js file in the `<head>` of html 

```
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>experiment</title>

   <script src="https://unpkg.com/jspsych@7.2.1"></script>
    <link href="https://unpkg.com/jspsych@7.2.1/css/jspsych.css" rel="stylesheet" type="text/css" />

<!-- plugin you used in your experiment  -->
   <script src="https://unpkg.com/@jspsych/plugin-html-keyboard-response@1.1.0"></script> 

<!-- the relative file location of your cognition script-->
    <script src =".\cognition_script.js"></script> 
</head>
<body>
    
</body>
</html>
```

**Reply by lana-mrkonja** [reply]
Thank you @rankiejiang ! Your answer worked, however I had to put the cognition js file in the `<body>` part for it to work throughout the entire task for some reason (I also had to load jspsych 6.2 version which the original code was written in, not sure if that has anything to do with the `<body>` vs `<head>` issue)
